pages/admin/user/[id].tsx

```diff
@@ -305,27 +312,28 @@ const EditUser: React.FunctionComponent<EditUserProps> = ({
 };
 
 const UserProfile: React.FunctionComponent = () => {
-  const [user, setUser] = useState<IUser>();
+  const [user, setUser] = useState<PUser>();
   const [isEditing, setIsEditing] = useState(false);
   const [isDeleting, setIsDeleting] = useState(false);
   const router = useRouter();
   const { id } = router.query;
 
   useEffect(() => {
     const getUser = async (): Promise<void> => {
-      const response = await fetch(`/api/admin/users/${id}`, {
+      const response = await fetch(`/api/admin/users/menteed/${id}`, {
```

Comment:
One thing I'd suggest here: instead of creating an API request for this specific use case (which doesn't appear to be reused anywhere else), why not specify the data needs of this page within a `getServerSideProps` function?

It seems to me that the data needs here would be:

* A user with `roles` included (we have this defined in /api/admin/users/readOne — consider breaking this out into its own function that doesn't require `req` and `res`, and reusing it then in the gSSP for this page)
* The users found in the `relatedPlayerIds` for all the user's `roles`. You can directly query Prisma for this, using `prisma.user.findMany({ where: { id: { in: user.defaultRole.relatedPlayerIds } } })`

You could choose how to specify these two data needs as props, but it might save us from maintaining another API route (by composing the logic of an existing one) and remove a roundtrip from the client. 

I also understand that this much moving around is a pretty big ask, so feel free to push back. I think my biggest hesitation here is the creation of a new general type "`PUser`" that fits this very specific use case where I think we can make it more of a specific type used within this page only.

Maybe a lower-cost way of addressing this concern is to create a specific type for the /menteed API response (we could call it something like `RelatedPlayersList` and export it from /api/admin/users/menteed/[id].ts) and use that in this response (as opposed to PUser/DashbordRole). 

Let me know what you think. I'm open to alternate approaches to addressing this concern.

Reply:
Thank you so much for the input Ethan! I'll go ahead and try making a getServerSideProps function and refactor from there.